content/docs/concepts/architecture.md

```diff
@@ -92,19 +90,23 @@ to 'refresh' the {{<color "#6A00FF" >}}vWire{{< /color >}}.  If a {{<color "#6A0
 
 A {{<color "#0050EF" >}}Client{{< /color >}} may use a 'MonitorConnection' streaming GRPC call to NSM to get updates on the status of a {{<color "#6A00FF" >}}vWire{{< /color >}} it has to a Network Service.
 
-## Registries
+## Registry
+
+The Network Service Registry (NSR) component tracks Network Services and Network Service {{<color "#008A00" >}}Endpoints{{</color>}}. It can use either in-memory volatile storage or Kubernetes custom resource storage.
 
-As with any other Mesh, Network Service Mesh has Network Service Registries (NSR) in which Network Services and Network Service {{<color "#008A00" >}}Endpoint{{< /color >}}s are registered
+Clients and Endpoints don't connect directly to the Registry; they connect to the Network Service Manager which is a proxy for the registry. This simplifies NSM's internal data flows since Clients and Endpoints need to connect to the Manager anyway.
```

Comment:
Your observation is absolutely true about the K8s implementation.  The indirection though is not a requirement in NSM.

Typically speaking, NSM utilizes a 'chain of responsibility' pattern.   (see [slides starting here](https://docs.google.com/presentation/d/154iByYvSI-g10RZRg6_xO0xBo-zpJopf/edit#slide=id.p16) ).

The chain of responsibility involves a 'proxy' for the NSM API (both the Request/Close/Monitor API and also for the Registry APIs) that assist in getting your vWire to/from an environment.  The NSMgr/Forwarder in K8s plays that role for a 'Node'.  

The local Pod literally can't handle getting a vWire to itself: it's not privleged, it has no access to getting a tunnel in or out of the Node.  The NSMgr/Forwarder act as a proxy to provide those functions to the Pod tranparently.  So the Pod reasons about what it wants, local mechnisms like kernel interface or memif interface, and leaves reasoning about how to get in or out of the Node to the NSMgr/Forwarder.

The same plays out getting in or out of the K8s Cluster.  The NSMgrproxy handles issues like translating between NodeIP and ExternalNodeIP etc.

This architecture makes it possible to further handle getting to/from other 'containing environments' like an on-prem DC, cloud VPC, etc by introducing a new proxy into the chain.

Please also note that a proxy may handle control plane, dataplane, or both.  The NSMgr/Forwarder end up touching both the controlplane and dataplane - changing control plane messaging to replace the 'local mechanism' of kernel or memif with a 'remote mechanism' like vxlan or wireguard (among other changes) - but also creating part of the dataplane as well (the cross connect in the Forwarder between the kernel interface or memif dropped into the Pod and the tunnel leaving the Node.

NSMgrproxy by contrast is only really proxying the control plane - it jiggers the IP address parameters of whatever remote mechanism is in use, but not setting up anything that will touch the data plane.

This ability to separate or combine control plane and data plane at each hop is... extremely handy :)

Reply:
Oh... and one other thing to be aware about wrt Registries: each Registry serves a domain.   While NSM generally has a 'Registry' domain local to each K8s cluster as currently implemented, you can also 'floating registries' not associated with any cluster.  To be consumable outside a local environment, Registries use SRV records for a DNS domain.  So if I want a Registry for the sales.example.com domain I would simply provide an SRV record pointing to its registry and then any workload that can then request a network service ('myns' for example) as myns@sales.example.com and, presuming the registry is willing to talk to the workload and the NSEs/Proxies are willing to accept the Request... get a vWire strung to them.

The same is true for NSEs.  An NSE can register with any Registry Domain that will have it (ie, authz).

Reply:
This is great info, thank you! Much of this is covered already but we need to add sections for the NSMgr and Forwarder to explain their roles. I think that might be out of scope of this PR, though.

Reply:
Definitely out of scope of this PR.  Also, the NSMgr/Forwarder are conceptually local to the K8s implementation.  Other environments may (or may not) have those particular components.

Reply:
On that note, I just pushed a small change to explain about the general/k8s-specific architecture.